src/main/java/dataMgmt/MetadataManager.java

```diff
@@ -156,6 +184,29 @@ public void fileMoved(long fileID, String newFilePath) {
         fileIDtoFilePath.put(fileID, newFilePath);
     }
 
+    public void fileDeleted(long fileID) {
+        String filePath = fileIDtoFilePath.get(fileID);
+        if (filePath != null){
+            fileMetadataMap.remove(filePath);
+        }
+        fileIDtoFilePath.remove(fileID);
+
+        long id = fileIDtoProjectID.get(fileID);
+        fileIDtoProjectID.remove(fileID);
```

Comment:
again, what happens if it missed? we also need to null check it before we remove it or it will throw an exception.